src/portable/portable.c

```diff
@@ -1035,10 +1035,14 @@ static bool marker_matches_image(const char *marker, const char *name_or_path) {
         a = last_path_component(marker);
 
         if (image_name_is_valid(name_or_path)) {
-                const char *e;
+                const char *e, *underscore;
 
                 /* We shall match against an image name. In that case let's compare the last component, and optionally
-                 * allow either a suffix of ".raw" or a series of "/". */
+                 * allow either a suffix of ".raw" or a series of "/".
+                 * But allow matching on a differen version of the same image, when a "_" is used as a separator. */
+                underscore = strchr(name_or_path, '_');
+                if (underscore)
+                        return memcmp(a, name_or_path, underscore - name_or_path) == 0;
```

Comment:
`strneq(a, name_or_path, underscore - name_or_path)`.

Hmm, shouldn't this match be only done for the last component of the path? Matching on `_` in directories seems wrong.

Reply:
> Hmm, shouldn't this match be only done for the last component of the path? Matching on _ in directories seems wrong

image_name_is_valid already enforces that name_or_path is not a path, but a single name - ie, without "/" anywhere.